src/core/src/Eryph.VmManagement/Storage/VMDriveStorageSettings.cs

```diff
@@ -15,25 +19,29 @@ public class VMDriveStorageSettings
 
 
         public static EitherAsync<Error, Seq<VMDriveStorageSettings>> PlanDriveStorageSettings(
-            VmHostAgentConfiguration vmHostAgentConfig, CatletConfig config, VMStorageSettings storageSettings)
+            VmHostAgentConfiguration vmHostAgentConfig,
+            CatletConfig config,
+            VMStorageSettings storageSettings,
+            Func<string, EitherAsync<Error, Option<VhdInfo>>> getVhdInfo)
         {
-            return config.Drives
-                .ToSeq().MapToEitherAsync((index, c) =>
-                    FromDriveConfig(vmHostAgentConfig, storageSettings, c, index).ToEither()).ToAsync();
+            return config.Drives.ToSeq()
+                .Map((index, c) => FromDriveConfig(vmHostAgentConfig, storageSettings, c, getVhdInfo, index))
+                .ToSeq().SequenceSerial();
         }
 
-        public static EitherAsync<Error, VMDriveStorageSettings> FromDriveConfig(
+        private static EitherAsync<Error, VMDriveStorageSettings> FromDriveConfig(
             VmHostAgentConfiguration vmHostAgentConfig,
             VMStorageSettings storageSettings,
             CatletDriveConfig driveConfig,
+            Func<string, EitherAsync<Error, Option<VhdInfo>>> getVhdInfo,
             int index)
         {
-            const int
-                controllerNumber = 0; //currently this will not be configurable, but keep it here at least as constant
-            var controllerLocation =
-                index; //later, when adding controller config support, we will have to add a logic to 
-            //set location relative to the free slots for each controller                   
-
+            //currently this will not be configurable, but keep it here at least as constant
+            const int controllerNumber = 0;
+            
+            //later, when adding controller config support, we will have to add a logic to 
+            //set location relative to the free slots for each controller   
+            var controllerLocation = index;
 
             //if it is not a vhd, we only need controller settings
             if (driveConfig.Type.HasValue && driveConfig.Type != CatletDriveType.VHD)
```

Comment:
For PHD I agree, however when we add config validation it should also be rejected earlier here https://github.com/eryph-org/eryph/blob/447576ea6e3488269bf77a2232f1eaf00afbe400/src/modules/src/Eryph.Modules.Controller/Operations/ValidateCatletConfigCommandHandler.cs#L85 like on https://github.com/eryph-org/eryph/blob/main/src/modules/src/Eryph.Modules.Controller/Networks/NetworkConfigValidator.cs

But validation for catlet config its it own topic and out of scope for this PR, so just drop it on agent as not implemented as suggested. 

For Shared VHD I added a commit that implements the missing part for VHD Sets. I actually have tested it in the past, but the feature was lost when switching to differencial disks as default. 

Shared VHD will also not work on our dev machines, so its a bit tricky to test. I'm now also trying to update the converge unit test, please check if you also have to update the unit tests you have added for this PR.

Reply:
One additional comment for PHD: 
The idea was to use the Source field in this case (thats why it is prefixed with "gene:" for genepool references 
So PHD is just a directly attached drive with source set to a physical drive. However this is typical only used in Fiber Channel drive setups, so its ok to put it on the roadmap for some time.

Reply:
After thinking about it, I changed it a bit:
For clustered shared disks there are two concepts: shared vhd and vhd sets. Originally only shared vhds were supported, but I added the logic for VHDSet (since shared vhds are deprecated in most cases). 
But shared VHDs are not so complicated to add, as they are just a flag on Add-VMHardDiskDrive.

Thefore I added VHDSet as catlet drive type in config model and updated the code to support both shared vhds and vhd sets.